contracts/mocks/MockBosonRouter.sol

```diff
@@ -232,6 +240,69 @@ contract MockBosonRouter is
         );
 
         emit LogOrderCreated(tokenIdSupply, msg.sender, metadata[5], TKNTKN);
+
+        return tokenIdSupply;
+    }
+
+    function requestCreateOrderTKNTKNWithPermit(
```

Comment:
This file looks like a copy and paster of the actual Boson Router. I don't understand the purpose of this, perhaps I am missing something...

I know the pattern of creating mock contracts to inherit from contract exposing functionality in an abstract contract (e.g.), but I am not sure what is going on here.

Reply:
I think @hswopeams can answer this. As far as I know is for testing some egde error case.

Reply:
This is used for a specific test case. It tests an error situation in the VoucherKernel, but the function in the VoucherKernel being tested can only be called from the BosonRouter. In order to test it, I had to create a separate copy of the BosonRouter that passes bad data to the VocherKernel to test that it reverts. It wasn't possible to call the VoucherKernel function directly and pass bad data to test that it reverts. The MockBosonRouter has been modified to pass bad data to `IVoucherKernel(voucherKernel).createPaymentMethod` from the `requestCreateOrderETHETH` function